website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -12,81 +12,166 @@ description: |-
 
 # Custom Database Secrets Engines
 
+~> The interface for custom database plugins has changed in Vault 1.6. Vault will
+   continue to recognize the now deprecated version of this interface for some time.
+   If you are using a plugin with the deprecated interface, you should upgrade to the
+   newest version. See [Upgrading database plugins](#upgrading-database-plugins)
+   for more details.
+
+~> **Advanced topic!** Plugin development is a highly advanced topic in Vault,
+   and is not required knowledge for day-to-day usage. If you don't plan on writing
+   any plugins, we recommend not reading this section of the documentation.
+
 The database secrets engine allows new functionality to be added through a
 plugin interface without needing to modify vault's core code. This allows you
 write your own code to generate credentials in any database you wish. It also
 allows databases that require dynamically linked libraries to be used as plugins
 while keeping Vault itself statically linked.
 
-~> **Advanced topic!** Plugin development is a highly advanced topic in Vault,
-and is not required knowledge for day-to-day usage. If you don't plan on writing
-any plugins, we recommend not reading this section of the documentation.
-
 Please read the [Plugins internals](/docs/internals/plugins) docs for more
 information about the plugin system before getting started building your
 Database plugin.
 
 ## Plugin Interface
 
-All plugins for the database secrets engine must implement the same simple interface.
+All plugins for the database secrets engine must implement the same interface. This interface
+is found in `sdk/database/dbplugin/v5/database.go`
 
 ```go
 type Database interface {
+	// Initialize the database plugin. This is the equivalent of a constructor for the
+	// database object itself.
+	Initialize(ctx context.Context, req InitializeRequest) (InitializeResponse, error)
+
+	// NewUser creates a new user within the database. This user is temporary in that it
+	// will exist until the TTL expires.
+	NewUser(ctx context.Context, req NewUserRequest) (NewUserResponse, error)
+
+	// UpdateUser updates an existing user within the database.
+	UpdateUser(ctx context.Context, req UpdateUserRequest) (UpdateUserResponse, error)
+
+	// DeleteUser from the database. This should not error if the user didn't
+	// exist prior to this call.
+	DeleteUser(ctx context.Context, req DeleteUserRequest) (DeleteUserResponse, error)
+
+	// Type returns the Name for the particular database backend implementation.
+	// This type name is usually set as a constant within the database backend
+	// implementation, e.g. "mysql" for the MySQL database backend. This is used
+	// for things like metrics and logging. No behavior is switched on this.
```

Comment:
```suggestion
	// for things like metrics and logging.
```

I think the last bit is applicable now, but hard to guarantee/enforce.

Reply:
I copy pasted this from the current code to keep consistency. This function has never had any behavior based on it as it is just used as an identifier.

Reply:
Fair enough. I guess we do control when and where we call this, so as long as we are mindful of its usage then this may remain to be the case.